includes/bp-members/classes/class-bp-members-endpoints.php

```diff
@@ -24,27 +55,175 @@ public function __construct() {
+				'callback'            => [ $this, 'get_items' ],
+				'permission_callback' => [ $this, 'get_items_permissions_check' ],
 				'args'                => $this->get_collection_params(),
-			),
-			'schema' => array( $this, 'get_public_item_schema' ),
-		) );
+			],
+			'schema' => [ $this, 'get_public_item_schema' ],
+		] );
 
-		register_rest_route( $this->namespace, '/' . $this->rest_base . '/(?P<id>[\d]+)', array(
-			array(
+		register_rest_route( $this->namespace, '/' . $this->rest_base . '/(?P<id>[\d]+)', [
+			[
 				'methods'             => WP_REST_Server::READABLE,
-				'callback'            => array( $this, 'get_item' ),
-				'permission_callback' => array( $this, 'get_item_permissions_check' ),
-				'args'                => array(
-					'context' => $this->get_context_param( array( 'default' => 'view' ) ),
-				),
-			),
-			'schema' => array( $this, 'get_public_item_schema' ),
-		) );
+				'callback'            => [ $this, 'get_item' ],
+				'permission_callback' => [ $this, 'get_item_permissions_check' ],
+				'args'                => [
+					'context' => $this->get_context_param( [ 'default' => 'view' ] ),
+				],
+			],
+			'schema' => [ $this, 'get_public_item_schema' ],
+		] );
+
+		register_rest_route( $this->namespace, '/' . $this->rest_base . '/(?P<id>[\d]+)', [
+			[
+				'methods'             => WP_REST_Server::EDITABLE,
+				'callback'            => [ $this, 'update_item' ],
+				'permission_callback' => [ $this, 'update_item_permissions_check' ],
+				'args'                => $this->get_endpoint_args_for_item_schema( WP_REST_Server::EDITABLE )
+			]
+		] );
+	}
+
+	/**
+	 * Updates a user profile.
+	 *
+	 * @since 0.0.1
+	 * @access public
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 *
+	 * @return WP_Error|WP_REST_Response Response object on success, or error object on failure.
+	 */
+	public function update_item( $request ) {
+		$id = (int) $request['id'];
+
+		$user = get_user_by( "id", $id );
+
+		if ( empty( $user ) ) {
+			return new WP_Error( 'rest_user_invalid_id', __( 'Invalid user ID.' ), [ 'status' => 404 ] );
+		}
+
+		$prepared_args = $this->prepare_item_for_database( $request );
+
+		if ( is_wp_error( $prepared_args ) ) {
+			return $prepared_args;
+		}
+
+		if ( array_key_exists( self::FIELD_NAME_API, $prepared_args ) ) {
+			xprofile_set_field_data( self::FIELD_NAME_XPROFILE, $id, $prepared_args[ self::FIELD_NAME_API ] );
+		}
+		if ( array_key_exists( self::FIELD_ABOUT_ME_API, $prepared_args ) ) {
+			xprofile_set_field_data( self::FIELD_ABOUT_ME_XPROFILE, $id, $prepared_args[ self::FIELD_ABOUT_ME_API ] );
+		}
+		if ( array_key_exists( self::FIELD_GENDER_API, $prepared_args ) ) {
+			xprofile_set_field_data( self::FIELD_GENDER_XPROFILE, $id, $prepared_args[ self::FIELD_GENDER_API ] );
+		}
+		if ( array_key_exists( self::FIELD_DOB_API, $prepared_args ) ) {
+			xprofile_set_field_data( self::FIELD_DOB_XPROFILE, $id, $prepared_args[ self::FIELD_DOB_API ] );
```

Comment:
Can you elaborate on "correct?"  - there is sanitization/format checking happening in the prepare for db function